examples/audio/uk_ireland_accent_recognition.py

```diff
@@ -0,0 +1,664 @@
+
+    x = keras.layers.Dense(256, activation="relu", name="dense_1")(inputs)
+    x = keras.layers.Dropout(0.15, name="dropout_1")(x)
+
+    x = keras.layers.Dense(384, activation="relu", name="dense_2")(x)
+    x = keras.layers.Dropout(0.2, name="dropout_2")(x)
+
+    x = keras.layers.Dense(192, activation="relu", name="dense_3")(x)
+    x = keras.layers.Dropout(0.25, name="dropout_3")(x)
+
+    x = keras.layers.Dense(384, activation="relu", name="dense_4")(x)
+    x = keras.layers.Dropout(0.2, name="dropout_4")(x)
+
+    outputs = keras.layers.Dense(len(class_names), activation="softmax", name="ouput")(
+        x
+    )
+
+    model = keras.Model(inputs=inputs, outputs=outputs, name="accent_recognition")
+
+    auc = keras.metrics.AUC(name="auc")
+
+    model.compile(
+        optimizer=keras.optimizers.Adam(learning_rate=1.9644e-5),
+        loss=keras.losses.CategoricalCrossentropy(),
+        metrics=["accuracy", auc],
+    )
+
+    return model
+
+
+model = build_and_compile_model()
+model.summary()
+
+"""
+## Class Weights Calculation
+
+Since the dataset is quite unbalanced, we wil use class_weight during training.
+Getting the class weights is a little tricky because even though we know the number of audio files for each class, it does not represent the number of samples for that class since Yamnet transforms each audio file into multiple audio samples of 0.96 seconds each.
+So every audio file will be split into a number of samples that is proportional to its length.
+Therefore, to get those weights, we have to calculate the number of samples for each class after preprocessing through Yamnet.
+"""
+
+class_counts = tf.zeros(shape=(len(class_names),), dtype=tf.int32)
+
+for X, y in iter(train_ds):
+    class_counts = class_counts + tf.math.bincount(
+        tf.cast(tf.math.argmax(y, axis=1), tf.int32), minlength=len(class_names)
+    )
+
+class_weight = {
+    i: tf.math.reduce_sum(class_counts).numpy() / class_counts[i].numpy()
+    for i in range(len(class_counts))
+}
+
+print(class_weight)
+
+"""
+## Callbacks
+
+A couple of Keras callbacks in order to:
+*   Stop whenever the validation AUC stops improving
+*   Save the best model
+*   Call TensorBoard in order to later view the training and validation logs
+"""
+
+early_stopping_cb = keras.callbacks.EarlyStopping(
+    monitor="val_auc", patience=10, restore_best_weights=True
+)
+
+model_checkpoint_cb = keras.callbacks.ModelCheckpoint(
+    MODEL_NAME + ".h5", monitor="val_auc", save_best_only=True
+)
+
+tensorboard_cb = keras.callbacks.TensorBoard(
+    os.path.join(os.curdir, "logs", model.name)
+)
+
+callbacks = [early_stopping_cb, model_checkpoint_cb, tensorboard_cb]
+
+"""
+## Training
+"""
+
+history = model.fit(
+    train_ds,
+    epochs=EPOCHS,
+    validation_data=valid_ds,
+    class_weight=class_weight,
+    callbacks=callbacks,
+    verbose=2,
+)
+
+"""
+## Results
+"""
+
+"""
+### TensorBoard
+"""
+
+"""
+```
+%reload_ext tensorboard
```

Comment:
Don't use TensorBoard for visualization since we cannot display this on the webpage. You could use matplotlib to display training/validation curves.